src/components/General/Detail.vue

```diff
@@ -0,0 +1,289 @@
+
+export default {
+  components: {
+    ConversationMessage,
+    ConversationCompose,
+    ProfilePicture,
+    RandomArt,
+    QBtn,
+    QInfiniteScroll,
+    QSpinnerDots,
+    QList,
+    QAlert,
+    QItem,
+    QIcon,
+    QTooltip,
+    QScrollArea,
+    QToolbar,
+    QToolbarTitle,
+    QResizeObservable,
+    QScrollObservable,
+  },
+  data () {
+    return {
+      newestMessageId: -1,
+      oldestMessageId: -1,
+      scrollPositionFromBottom: 0,
+      fetchingMore: false,
+    }
+  },
+  computed: {
+    ...mapGetters({
+      user: 'auth/user',
+      pickup: 'detail/pickup',
+      conversation: 'detail/conversation',
+      away: 'presence/toggle/away',
+    }),
+    hasLoaded () {
+      if (!this.pickup || !this.conversation) return false
+      const s = this.conversation.fetchStatus
+      return !s.pending && !s.hasValidationErrors
+    },
+    messagePrompt () {
+      if (this.conversation.messages.length > 0) {
+        return this.$t('WALL.WRITE_MESSAGE')
+      }
+      else {
+        return this.$t('WALL.WRITE_FIRST_MESSAGE')
+      }
+    },
+    data () {
+      return this.conversation
+    },
+    reversedMessages () {
+      if (!this.conversation) return []
+      // eslint-disable-next-line vue/no-side-effects-in-computed-properties
+      return [...this.conversation.messages].reverse()
+    },
+  },
+  watch: {
+    away (away) {
+      if (!this.away) this.markRead(this.newestMessageId)
+    },
+    hasLoaded (hasLoaded) {
+      if (hasLoaded) this.scrollToBottom()
+    },
+    'pickup.isUserMember' (isUserMember) {
+      if (!isUserMember) this.$emit('close')
+    },
+    'conversation.id' (id) {
+      Object.assign(this, {
+        newestMessageId: -1,
+        oldestMessageId: -1,
+        scrollPositionFromBottom: 0,
+      })
+    },
+    'conversation.messages' (messages) {
+      if (!messages || messages.length === 0) return
+      // Jump to bottom when new messages added
+      const newNewestMessageId = messages[0].id
+      if (this.newestMessageId !== newNewestMessageId) {
+        this.scrollToBottom()
+        this.newestMessageId = newNewestMessageId
+        if (!this.away) this.markRead(this.newestMessageId)
+      }
+      // Retain position when old message added
+      const newOldestMessageId = messages[messages.length - 1].id
+      if (this.oldestMessageId !== newOldestMessageId) {
+        this.oldestMessageId = newOldestMessageId
+        this.$nextTick(() => {
+          this.restoreScrollPosition()
+        })
+      }
+    },
+  },
+  methods: {
+    ...mapActions({
+      send: 'conversations/send',
+      mark: 'conversations/mark',
+      toggleEmailNotifications: 'conversations/maybeToggleEmailNotifications',
+      toggleReaction: 'conversations/toggleReaction',
+      fetchMore: 'conversations/fetchMore',
+    }),
+    markRead (messageId) {
+      if (!this.conversation) return
+      this.mark({
+        id: this.conversation.id,
```

Comment:
During testing, I sometimes see requests to `/api/conversations/undefined/mark/` when opening the pickup chat while the page is loading.
The check above should prevent it, but maybe it doesn't? Could need some debugging, or could also be ignored...